libs/zk/include/nil/crypto3/zk/commitments/polynomial/lpc.hpp

```diff
@@ -421,6 +523,60 @@ namespace nil {
                         eval_storage_type z;
                         typename basic_fri::proof_type fri_proof;
                     };
+
+                    // Represents an initial proof, which must be created for each of the N provers.
+                    struct initial_proof_type {
+                        bool operator==(const initial_proof_type &rhs) const {
+                            return initial_fri_proof == rhs.initial_fri_proof && z == rhs.z;
+                        }
+
+                        bool operator!=(const initial_proof_type &rhs) const {
+                            return !(rhs == *this);
+                        }
+
+                        eval_storage_type z;
+                        typename basic_fri::initial_proofs_batch_type initial_fri_proof;
+                    };
+
+                    // Represents a round proof, which must be created just once on the main prover.
+                    struct round_proof_type {
+                        bool operator==(const round_proof_type &rhs) const {
+                            return fri_round_proof == rhs.fri_round_proof &&
+                                fri_commitments_proof_part == rhs.fri_commitments_proof_part;
+                        }
+
+                        bool operator!=(const round_proof_type &rhs) const {
+                            return !(rhs == *this);
+                        }
+
+                        // We have a single round proof for checking that F(X) is a low degree polynomial.
+                        typename basic_fri::round_proofs_batch_type fri_round_proof;
+
+                        // Contains fri_roots and final_polynomial that correspond to the polynomial F(x).
+                        typename basic_fri::commitments_part_of_proof fri_commitments_proof_part;
+                    };
+
+                    // A single instance of this class will store all the LPC proofs for a group of provers
+                    // when aggregated FRI is used.
+                    struct aggregated_proof_type {
+                        bool operator==(const aggregated_proof_type &rhs) const {
+                            return round_proofs == rhs.round_proofs &&
+                                intial_proofs_per_prover == rhs.intial_proofs_per_prover &&
+                                proof_of_work == rhs.proof_of_work;
+                        }
+
+                        bool operator!=(const proof_type &rhs) const {
+                            return !(rhs == *this);
+                        }
+
+                        // We have a single round proof for checking that F(X) is a low degree polynomial.
+                        round_proof_type round_proofs;
+
+                        // For each prover we have an initial proof.
+                        std::vector<initial_proof_type> intial_proofs_per_prover;
```

Comment:
`std::vector<lpc_proof_type> lpc_proofs_per_prover` -- vector size is similar to provers amount